src/tasks/WasmAppBuilder/LogAdapter.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace WasmAppBuilder {
+    public sealed class LogAdapter {
+        public bool HasLoggedErrors {
+            get => Helper?.HasLoggedErrors ?? _HasLoggedErrors;
+        }
+        private bool _HasLoggedErrors;
+        private TaskLoggingHelper? Helper;
+        private TextWriter? Output, ErrorOutput;
+
+        public LogAdapter (TaskLoggingHelper helper) {
+            Helper = helper;
+            Output = null;
+            ErrorOutput = null;
+        }
+
+        public LogAdapter () {
+            Helper = null;
+            Output = Console.Out;
+            ErrorOutput = Console.Error;
+        }
+
+        private static string AutoFormat (string s, object[] o) {
+            if ((o?.Length ?? 0) > 0)
+                return string.Format(s!, o!);
+            else
+                return s;
+        }
+
+        public void LogMessage (string s, params object[] o) {
+            Helper?.LogMessage(s, o);
+            Output?.WriteLine(AutoFormat(s, o));
+        }
+
+        public void LogMessage (MessageImportance mi, string s, params object[] o) {
```

Comment:
Should this also drop `Log` prefix like the other methods?

Reply:
I left it this way so existing code would keep working, I can go through and rename all the call sites if you want (it looks like there are about 30)

Reply:
oh then I'm fine with either.